PyU4V/tools/openstack/migrate_utils.py

```diff
@@ -0,0 +1,911 @@
+        """Get the list of volumes from the storage group.
+
+        :param storage_group_name: the storage group name -- str
+        :returns: volume list -- list, create volume -- boolean
+        """
+        create_volume_flag = False
+        volume_list = self.conn.provisioning.get_volumes_from_storage_group(
+            storage_group_name)
+        print_str = 'There are %d volume in storage group %s.'
+        self.smart_print(print_str, DEBUG, len(volume_list),
+                         storage_group_name)
+        txt = ('Do you want to migrate all %s volumes: Y/N: '
+               % len(volume_list))
+        txt_out = self.input(txt)
+        if self.check_input(txt_out, 'Y'):
+            # Move list of devices from old to new masking view
+            print_str = ('Moving all volumes between source '
+                         'and target storage groups.')
+            self.smart_print(print_str, DEBUG)
+            create_volume_flag = True
+        else:
+            volume_list, create_volume_flag = self.choose_subset_volumes(
+                storage_group_name, volume_list)
+
+        return volume_list, create_volume_flag
+
+    def choose_storage_group(
+            self, masking_view_name, child_storage_group_list, port_group,
+            initiator_group, revert=False):
+        """Choose a child storage group from the list.
+
+        :param masking_view_name: the masking view name -- str
+        :param child_storage_group_list: the storage group list -- list
+        :param port_group: The port group name -- str
+        :param initiator_group: the initiator group name -- str
+        :param revert: revert back -- boolean
+        :returns: element details -- dict, child storage group name -- str
+        """
+        element_dict = {}
+        child_storage_group_name = None
+        for child_storage_group_name in child_storage_group_list:
+            txt = ('Which storage group do you want to migrate:\n\t '
+                   '%s. Y/N: ' % child_storage_group_name)
+            txt_out = self.input(txt)
+            if 'Y' in txt_out:
+                # Compile the new names of the SGs and MV
+                element_dict = self.compile_new_element_names(
+                    masking_view_name, port_group, initiator_group,
+                    child_storage_group_name, revert)
+                return element_dict, child_storage_group_name
+        return element_dict, child_storage_group_name
+
+    def validate_masking_view(self, masking_view_name, revert=False):
+        """Validate the masking view string.
+
+        :param masking_view_name: masking view name -- str
+        :param revert: revert back -- boolean
+        :returns: validate flag -- boolean
+        """
+        if re.search('^OS-', masking_view_name):
+            try:
+                if revert:
+                    return True if self.get_masking_view_component_dict(
+                        masking_view_name, sys.argv[1]) else False
```

Comment:
sys.argv[1] needs converted to a boolean dependent on user input string, expected bool got str